### PR TITLE
Make crane possible to use without flakes

### PR DESCRIPTION
**Pull request by adisbladis**
I'd like to use Crane but I have no interest in starting to use Flakes.

With this change it's possible to use Crane without flakes like:
``` nix
{ lib ? pkgs.lib
, pkgs ? import <nixpkgs> { }
}:

let
  craneLib = import ../crane { inherit pkgs; };

  src = lib.cleanSource ./.;

  cargoArtifacts = craneLib.buildDepsOnly {
    inherit src;
  };

in craneLib.buildPackage {
  inherit cargoArtifacts src;
}
```

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -1,11 +1,6 @@
-let
-  lock = builtins.fromJSON (builtins.readFile ./flake.lock);
-  locked = lock.nodes.flake-compat.locked;
-  compat = fetchTarball {
-    url = "https://github.com/${locked.owner}/${locked.repo}/archive/${locked.rev}.tar.gz";
-    sha256 = locked.narHash;
-  };
+{ pkgs }:
```
Comment (ipetkov) [inline default.nix:1]: Given that this file is for explicitly avoiding using flakes, maybe it's worth defaulting to importing`<nixpkgs>` for convenience

```suggestion
{ pkgs ? import <nixpkgs> {} }:
```

File: lib/writeTOML.nix [hunk 2]
```diff
@@ -1,5 +1,13 @@
...
-, writeText
+{ writeText
+, remarshal
+, runCommand
 }:
 
-name: contents: writeText name ((toTOML contents) + "\n")
+name: contents: runCommand name
+{
+  contents = builtins.toJSON contents;
```
[the remark this answers is not in the document]
Comment (adisbladis) [reply lib/writeTOML.nix:8]: Yes it does.

**Comment by adisbladis** [comment]
I've added a check that uses the root level default.nix.

**Comment by adisbladis** [comment]
I have no idea why the example check is currently failing in CI. Exactly the same command works on a flakes-enabled machine I tried it on.

**Comment by ipetkov** [comment]
@adisbladis I'm able to reproduce it locally with:

```sh
cd examples/cross-rust-overlay
nix flake check --override-input crane ~/crane --override-input nixpkgs github:NixOS/nixpkgs/release-22.05
```

Seems like the process gets killed due to running out of memory, not sure what is causing it to get stuck

**Comment by ipetkov** [comment]
Ok seems like something weird is happening when Nix tries to resolve the `remarshal` input in `lib/writeTOML` on the 22.05 release branch. Changing the input to `nativeBuildInputs = [ pkgs.pkgsBuildBuild.remarshal ];` appears to resolve the infinite loop, no idea why (wonder if its a regression on the stable branch...)

**Comment by adisbladis** [comment]
Alright, your suggested fix applied.

**Pull request merged**